javascript/ql/src/semmle/javascript/dataflow/Configuration.qll

```diff
@@ -1136,6 +1155,17 @@ private predicate loadStep(
     parameterPropRead(f, succ, pred, prop, read, cfg) and
     reachesReturn(f, read, cfg, summary)
   )
+  or
+  // load in an immidiatly awaited function
```

Comment:
```suggestion
  // load from an immediately awaited function call
```

Reply:
In line with the above, it would be nice with a predicate like `FlowSteps::awaitReturnStep(returnExprNode, awaitExprNode)` connecting the `await` expression to the return expression in the body of the callee. Seems like it could be used in both places.

Reply:
> In line with the above, it would be nice with a predicate like FlowSteps::awaitReturnStep(returnExprNode, awaitExprNode) connecting the await expression to the return expression in the body of the callee. Seems like it could be used in both places.

A `awaitReturnStep` predicate doesn't fit, because in some cases the `returnExprNode` is computed in another predicate (e.g. `returnedPropWrite`).  

I managed to de-duplicate the implementations such that the "body" is shared between the async and non-async load/store steps.